LCD_digits/LCDNumberConverter.js

```diff
@@ -0,0 +1,41 @@
+let patterns = require('./Patterns').patterns;
+
+function cleanInput(number) {
+    if(typeof(number) === 'string' && !isNaN(number)) number = parseInt(number);
+    if(typeof(number) !== 'number') number = 0;
+    number = Math.trunc(number);
+    return number;
+}
+
+function buildLCDString(digits) {
+    let lines = ['','',''];
```

Comment:
Typically if you setup a variable so that you could build it during an array/collection loop and then return it you can replace the work with a reduce statement :

```javascript

function buildLCDString(digits) {
    return digits.reduce((lines, digit, i) => {
        let digitPattern = patterns[digit];
        lines[0] += digitPattern[0];
        lines[1] += digitPattern[1];
        lines[2] += digitPattern[2];
        if(i < digits.length-1) {
            lines[0] += ' ';
            lines[1] += ' ';
            lines[2] += ' ';
        }
        return lines;
    }, ['','',''])
    .join('\n');
}
```

you could also consider splitting up some of the body into methods

```javascript
function buildLCDString(digits) {
    return digits.reduce((lines, digit, i) => {
        addDigitsToLines(lines, patterns[digit]);
        if(i < digits.length-1) {
            paddLine(lines);
        }
        return lines;
    }, ['','',''])
    .join('\n');
}

const addDigitsToLines = (lines, digitPattern) => {
    for(let i = 0; i < 3; i++) {
        lines[i] += digitPattern[i];
    }
}

const paddLine = (lines) => {
    lines[0] += ' ';
    lines[1] += ' ';
    lines[2] += ' ';
}
```

This creates methods that modify the argument though. The only way to get around that, since that signifies a need for state, is to make a class that manages the 3x3 line grid. That may be overkill, so the functions will be fine as long as they indicate that they are modifying the array.